### PR TITLE
Expand Copter and Rover failsafes

### PR DESCRIPTION
**Pull request by Hwurzburg**

### CODE REVIEW

**Comment by Hwurzburg** [comment]
@rmackay9  added note at top. .....changed time to FS to 5second...you had mentioned that there might be a FS_GCS_TIMEOUT param...I cant seem to find it....

File: copter/source/docs/gcs-failsafe.rst [hunk 1]
```diff
@@ -3,8 +3,9 @@
 ============
 GCS Failsafe
 ============
+Significant improvements have been made in Copter-4.0 and later for Ground Control Failsafe .
 
-The Ground Station Control (GCS) failsafe controls how Copter will behave if contact with the GCS is lost.  The GCS failsafe monitors the time since the last mavlink heartbeat from the GCS.  If no heartbeat is received in over 2.5 seconds, the GCS failsafe event will trigger based on your parameter settings. Note that if no GCS is ever connected, the GCS failsafe will remain inactive regardless of parameter settings.
+The Ground Station Control (GCS) failsafe controls how Copter will behave if contact with the GCS is lost.  The GCS failsafe monitors the time since the last mavlink heartbeat from the GCS.  If no heartbeat is received in over 5 seconds or 2.0 seconds if the vehicle is being controlled by Joystick thru the GCS, the GCS failsafe event will trigger based on your parameter settings. Note that if no GCS is ever connected, the GCS failsafe will remain inactive regardless of parameter settings.
```
Comment (rmackay9) [inline copter/source/docs/gcs-failsafe.rst]: We should remove the mention of the GCS failsafe sometimes being 2sec.. but it's actually always 5sec

File: rover/source/docs/rover-failsafes.rst [hunk 2]
```diff
@@ -45,7 +45,7 @@ This failsafe is triggered if the vehicle stops receiving `heartbeat messages <h
 
 - set :ref:`FS_GCS_ENABLE <FS_GCS_ENABLE>` to "1" to enable this failsafe
 - if :ref:`FS_ACTION <FS_ACTION>` is "1", the vehicle will :ref:`RTL <rtl-mode>` to home, if "2" the vehicle will :ref:`Hold <hold-mode>`
-- use the transmitter's mode switch to re-take control of the vehicle in :ref:`Manual <manual-mode>` (or any other mode)
+- you must use the transmitter's mode switch to re-take control of the vehicle in :ref:`Manual <manual-mode>` (or any other mode), or change modes via the GCS iself, if communication is re-established.
```
Comment (rmackay9) [inline rover/source/docs/rover-failsafes.rst]: it's an existing issue but the pilot could also change the mode from the gcs.. but I suspect most users will figure this out anyway

Comment (Hwurzburg) [reply rover/source/docs/rover-failsafes.rst]: not sure what you mean....what is the existing issue? and it says that they pilot can change it out of the gcs fs mode with the gcs itself if communication is re-established with the gcs...is this incorrect?

File: copter/source/docs/gcs-failsafe.rst (hunk 1 above)
Comment (Hwurzburg) [reply copter/source/docs/gcs-failsafe.rst]: will fix on next update to PR

Comment (Hwurzburg) [reply copter/source/docs/gcs-failsafe.rst]: done

**Comment by Hwurzburg** [comment]
think I got everything now

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**